lib/evmone/instructions.hpp

```diff
@@ -252,118 +302,180 @@ inline void signextend(StackTop stack) noexcept
         const auto sign_ex = static_cast<uint64_t>(static_cast<int64_t>(sext_byte) >> 8);
 
         for (size_t i = 3; i > sign_word_index; --i)
-            x[i] = sign_ex;  // Clear extended words.
+            x.val[i] = sign_ex;  // Clear extended words.
+        
+        if (std::holds_alternative<Pure>(*stack[0].sval) && std::holds_alternative<Pure>(*stack[1].sval))
+            stack[1].sval = std::make_shared<SymbolicStackItem>(Pure {stack[1].val});
+        else
+            stack[1].sval = std::make_shared<SymbolicStackItem>(BinaryOp {BinOp::signextend, stack[0].sval, stack[1].sval});
```

Comment:
I think we need to update `stack[1]` also when `ext.val < 31`. I believe that we can simply move the update outside below the `if`.